client-runtime/src/main/java/com/microsoft/rest/v2/policy/RetryPolicy.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.policy;
+
+import com.microsoft.rest.v2.http.HttpRequest;
+import com.microsoft.rest.v2.http.HttpResponse;
+import rx.Single;
+import rx.functions.Func1;
+
+/**
+ * An instance of this interceptor placed in the request pipeline handles retriable errors.
+ */
+public final class RetryPolicy implements RequestPolicy {
+    /**
+     * Factory which instantiates RetryPolicy.
+     */
+    public static class Factory implements RequestPolicy.Factory {
+        private static final int DEFAULT_NUMBER_OF_ATTEMPTS = 3;
+        final int maxRetries;
+
+        /**
+         * Creates a Factory with the default number of retry attempts.
+         */
+        public Factory() {
+            maxRetries = DEFAULT_NUMBER_OF_ATTEMPTS;
+        }
+
+        /**
+         * Creates a Factory.
+         * @param maxRetries The maximum number of retries to attempt.
+         */
+        public Factory(int maxRetries) {
+            this.maxRetries = maxRetries;
+        }
+
+        @Override
+        public RequestPolicy create(RequestPolicy next) {
+            return new RetryPolicy(maxRetries, next);
+        }
+    }
+
+    private final int maxRetries;
+    private final RequestPolicy next;
+    private int tryCount = 0;
+
+    private RetryPolicy(int maxRetries, RequestPolicy next) {
+        this.maxRetries = maxRetries;
+        this.next = next;
+    }
+
+    private boolean shouldRetry(HttpResponse response) {
+        int code = response.statusCode();
+        //CHECKSTYLE IGNORE MagicNumber FOR NEXT 2 LINES
+        return tryCount < maxRetries
+                && (code == 408 || (code >= 500 && code != 501 && code != 505));
```

Comment:
You might be able to make checkstyle happy and make this code easier to read if you extract these into static final variables with names. I have to admit that I don't remember what a 505 status code means.